crowbar_framework/app/models/service_object.rb

```diff
@@ -822,8 +822,19 @@ def apply_role(role, inst, in_queue)
     # We also check that all nodes we'll require are in the ready state.
     #
 
-    # Initialize variables used in ensure at the end of the method
-    chef_daemon_nodes = []
+    nodes_without_admin = NodeObject.find_all_nodes.reject do |node|
+      node[:platform] == "windows" || node.admin?
+    end.map(&:name)
+
+    apply_locks = nodes_without_admin.map do |node|
```

Comment:
Ugh, this is horrible.  I've had to move the locking code later, because at this point we don't yet know which nodes are related to the proposal being applied.  And that means also moving the code which waits for any currently running chef-client runs to finish so that it's later too.  Due to the tangled monstrosity of `apply_role`, it's hard to know if I got it right - however I think I chose the best spot which is basically as early as possible and before anything significant gets changed.  The only thing which worries me is an inexplicable `role.save` very early on in `apply_role` - any ideas what that's for?

@vuntz I'm relying heavily on you for sanity checks here :-/